src/components/options/settings/importExport.svelte

```diff
@@ -0,0 +1,201 @@
+<script>
+import _ from "lodash";
+import beautify from "js-beautify";
+import BSN from "bootstrap.native/dist/bootstrap-native.esm.min.js";
+import Notiflix from "notiflix";
+import { createEventDispatcher } from "svelte";
+
+import ConfigFile from "../../../js/shared/config_file";
+import { EXPORT_FILE_NAME } from "../../../js/shared/constants";
+
+const dispatch = createEventDispatcher();
+
+let editModal = null;
+
+function closeModal() {
+  editModal.dispose();
+}
+
+async function exportToFile() {
+  // Generate the JSON
+  let data = await ConfigFile.generateJSONFile();
+
+  // Create a downloadable link with content. I have named the exported file to 'Settings.json'.
+  var downloadLink = document.createElement("a");
+  var blob = new Blob([JSON.stringify(data)], {
+    type: "text/plain;charset=utf-8",
+  });
+  var url = URL.createObjectURL(blob);
+  downloadLink.href = url;
+  downloadLink.download = EXPORT_FILE_NAME;
+
+  document.body.appendChild(downloadLink);
+  downloadLink.click();
+  document.body.removeChild(downloadLink);
+}
+
+function fileImported(event) {
+  let selectedFile = _.first(this.files);
+  var reader = new FileReader();
+
+  reader.onload = async function (event) {
+    await saveSearches(event.target.result);
+  };
+  reader.readAsText(selectedFile);
+
+  // Reset the value
+  event.target.value = "";
+}
+
+function importFromFile() {
+  document.getElementById("settings_fileInput").click();
+}
+
+async function openModal() {
+  // Update text Area
+  await updateJSONTextarea();
+  editModal = new BSN.Modal("#settings_editModal");
+  editModal.show();
+}
+
+async function saveModalChanges() {
+  var changes = document.getElementById("settings_json").value;
```

Comment:
I think it should be better to use [Svelte's bindings](https://svelte.dev/tutorial/bind-this) for this, so instead of doing `id="settings_json"` you do `bind:this={modalTextarea}` and instead of `document.getElementById("settings_json").value` you can just do `modalTextarea.value`.